### PR TITLE
Flat field added to apartment building

### PR DESCRIPTION
**Pull request by ogbeche77**
Hey Maintainers, I am new here and willing to contribute to this community.
I observed that this issue not been resolved, https://github.com/openstreetmap/iD/issues/7856 so I went ahead to implement the change as described by @TheAdventurer64.
A flat field was added to indicate how many flats are in an apartment building. If label needs to be reworded/rephrased, I will gladly do that again.
Cheers

### CODE REVIEW

File: data/presets/fields/flats.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "key": "flats",
```
Comment (quincylvania) [inline data/presets/fields/flats.json]: The key for this is actually `building:flats`. Could you change this and move the file to `data/presets/fields/building/flats.json`?

File: data/presets/fields/flats.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    "type": "number",
+    "minValue": 0,
+    "label": "Flats",
```
Comment (quincylvania) [inline data/presets/fields/flats.json]: So iD's default language is American English. Could we make the label "Units" instead?

**Review by quincylvania — CHANGES_REQUESTED** [review]
Looks pretty good, @ogbeche77, just a couple changes needed before we can merge.

File: data/presets/fields/flats.json (hunk 1 above)
Comment (ogbeche77) [reply data/presets/fields/flats.json]: Hi  @quincylvania 
Thank you for the feedback. I have made the changes as requested.

**Comment by ogbeche77** [comment]
Hi  @quincylvania 
Thank you for the feedback. I have made the changes as requested.

File: data/presets/fields/flats.json (hunk 1 above)
Comment (quincylvania) [reply data/presets/fields/flats.json]: Looks like you added the new file, but you'll also need to delete this file :)

File: data/presets/presets/building/apartments.json [hunk 3]
```diff
@@ -1,5 +1,9 @@
 {
     "icon": "maki-building",
+    "fields": [
+        "{building}",
+        "flats"
```
Comment (quincylvania) [inline data/presets/presets/building/apartments.json]: Since the preset file was moved to a folder, the new field ID needs to be `building/flats`.

Comment (ogbeche77) [reply data/presets/presets/building/apartments.json]: Hi Quincy, thanks for your patience :)
Changes have been made as requested

File: data/presets/presets/building/apartments.json [hunk 4]
```diff
@@ -1,5 +1,8 @@
 {
     "icon": "maki-building",
+    "fields": [
+        "building/flats"
```
Comment (quincylvania) [inline data/presets/presets/building/apartments.json:5]: Oh, you should also have `"{building}"` here the way you did before.

Comment (ogbeche77) [reply data/presets/presets/building/apartments.json:5]: So it should be `  "fields": [ "{building}"]` ? Without "flats"?  or ` "fields": [ "{building}","flats"]`  ?

I think I need more time to get more familiar with the code base :)

Comment (ogbeche77) [reply data/presets/presets/building/apartments.json:5]: Oh.. I think I figured it out. It should be `"fields": [
        "{building}",
        "building/flats"
    ]`
:)

**Review by quincylvania — APPROVED** [review]

**Comment by quincylvania** [comment]
@ogbeche77 Looks great, thanks for the contribution and for working through the technicalities! Welcome to iD development.

**Pull request merged**